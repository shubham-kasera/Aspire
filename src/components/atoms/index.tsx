import Chip from './Chip';
import ButtonComponent from './ButtonComponent';
import ToggleButton from './ToggleButton';

export {Chip, ButtonComponent, ToggleButton};
