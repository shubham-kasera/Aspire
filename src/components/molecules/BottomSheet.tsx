import React, {FC} from 'react';
import {Modalize} from 'react-native-modalize';
import {Platform, StyleProp, StyleSheet, ViewStyle} from 'react-native';
import {hp} from '../../styles/globalStyles';

interface BottomSheet {
  children?: React.ReactNode;
  childrenStyle?: StyleProp<ViewStyle>;
}

const BottomSheet: FC<BottomSheet> = props => {
  return (
    <Modalize
      alwaysOpen={Platform.OS == 'ios' ? hp('60%') : hp('64%')}
      modalTopOffset={hp('16%')}
      overlayStyle={styles.overlayStyle}
      modalStyle={styles.modalStyle}
      withHandle={false}
      childrenStyle={props.childrenStyle}
      scrollViewProps={{showsVerticalScrollIndicator: false}}
      children={props.children}
    />
  );
};

const styles = StyleSheet.create({
  modalStyle: {
    alignItems: 'center',
    borderTopLeftRadius: 24,
    borderTopRightRadius: 24,
  },
  overlayStyle: {
    backgroundColor: 'transparent',
  },
});

export default BottomSheet;
