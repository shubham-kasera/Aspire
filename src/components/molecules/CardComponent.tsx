import React, {FC} from 'react';
import {Image, View, Text, StyleSheet, Platform} from 'react-native';
import {flexRow, globalStyles, wp, hp} from '../../styles/globalStyles';
import {size, type} from '../../theme/fonts';

interface Card {
  cardHolderName: string;
  cardNum: string;
  expiryDate: string;
  cvv: string;
  dataVisible: boolean;
}

const CardComponent: FC<Card> = props => {
  return (
    <View style={[styles.CardWrapper, globalStyles.shadow4]}>
      <Image
        resizeMode="contain"
        style={styles.upperLogo}
        source={require('../../assets/images/AspireLogo.png')}
      />
      <View
        style={{
          marginTop: '2%',
          flex: 1,
          justifyContent: 'flex-end',
          paddingBottom: '10%',
        }}>
        <Text style={styles.cardHolderName}>{props.cardHolderName}</Text>
        <View style={[flexRow.between, styles.cardNumberWrapper]}>
          {[...props.cardNum].map((element, index) => {
            if (index < 15 && !props.dataVisible) {
              if (element != '-') {
                return <View key={index} style={styles.dotStyle} />;
              } else {
                return <View key={index} style={{marginHorizontal: 7}} />;
              }
            } else {
              return (
                <Text key={index} style={styles.cardNumber}>
                  {element != '-' ? element : ' '}
                </Text>
              );
            }
          })}
        </View>
        <View style={[flexRow.between, styles.subDetailsWrapper]}>
          <Text style={styles.subDetails}>Thru: {props.expiryDate}</Text>
          <View
            style={[
              flexRow.between,
              {width: '40%', justifyContent: 'flex-start'},
            ]}>
            <Text style={styles.subDetails}>CVV:</Text>
            {!props.dataVisible ? (
              <Text style={styles.cvvNum}>***</Text>
            ) : (
              <Text style={[styles.subDetails, {marginLeft: 5}]}>
                {props.cvv}
              </Text>
            )}
          </View>
        </View>
        <Image
          resizeMode="contain"
          style={styles.bottomLogo}
          source={require('../../assets/images/VisaLogo.png')}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  CardWrapper: {
    width: wp('90%'),
    height: Platform.OS == 'ios' ? hp('26%') : hp('29%'),
    backgroundColor: '#01D167',
    borderRadius: 12,
    paddingHorizontal: '5%',
    paddingVertical: '3%',
    flex: 1,
  },
  upperLogo: {
    alignSelf: 'flex-end',
    width: '22.5%',
  },
  cardHolderName: {
    color: '#fff',
    fontSize: size.font22,
    fontFamily: type.avenirBold,
  },
  cardNumberWrapper: {
    marginTop: '9%',
    width: '85%',
  },
  dotStyle: {
    backgroundColor: '#fff',
    width: 9,
    height: 9,
    borderRadius: 20,
  },
  cardNumber: {
    color: '#fff',
    fontSize: size.font14,
    fontFamily: type.avenirDemi,
  },
  subDetailsWrapper: {
    marginTop: '4%',
    width: '54%',
    height: '20%',
  },
  subDetails: {
    color: '#fff',
    fontSize: size.font13,
    fontFamily: type.avenirDemi,
  },
  cvvNum: {
    color: '#fff',
    fontSize: size.font22,
    fontFamily: type.avenirDemi,
    marginBottom: -9,
  },
  bottomLogo: {
    alignSelf: 'flex-end',
    width: '18%',
    position: 'absolute',
    bottom: 0,
  },
});

export default CardComponent;
