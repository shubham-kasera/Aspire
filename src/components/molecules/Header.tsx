import React, {FC} from 'react';
import {Image, View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import Feather from 'react-native-vector-icons/Feather';
import {size, type} from '../../theme/fonts';
import {wp, hp} from '../../styles/globalStyles';

interface Header {
  title?: string;
  leftIcon?: boolean;
  onBackPress?: () => void;
}

const Header: FC<Header> = props => {
  return (
    <View style={styles.headerWrapper}>
      {!props.leftIcon ? (
        <View style={styles.titleWrapper}>
          <Text style={styles.title}>{props.title}</Text>
        </View>
      ) : (
        <TouchableOpacity activeOpacity={1} onPress={props.onBackPress}>
          <Feather name={'chevron-left'} color="#fff" size={hp('3.5%')} />
        </TouchableOpacity>
      )}

      <Image
        resizeMode="contain"
        style={styles.logoStyle}
        source={require('../../assets/images/Logo.png')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  headerWrapper: {
    height: hp('7.5%'),
    width: wp('100%'),
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: '5%',
  },
  logoStyle: {
    width: '7.5%',
  },
  titleWrapper: {
    height: '100%',
    justifyContent: 'flex-end',
  },
  title: {
    fontSize: size.font24,
    fontFamily: type.avenirBold,
    color: '#fff',
    marginBottom: -10,
  },
});

export default Header;
