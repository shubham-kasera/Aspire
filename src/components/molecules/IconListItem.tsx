import React, {FC} from 'react';
import {Image, Platform, StyleSheet} from 'react-native';
import {List, Switch} from 'react-native-paper';
import {size, type} from '../../theme/fonts';
import {hp} from '../../styles/globalStyles';

interface IconListItem {
  title: string;
  desc?: string;
  logo: Object;
  showSwitch?: boolean;
  onPress?: () => void;
  isSwitchOn?: boolean;
  onValueChange?: () => void;
}

const IconListItem: FC<IconListItem> = props => {
  return (
    <List.Item
      title={props.title}
      description={props.desc}
      titleStyle={styles.titleStyle}
      descriptionStyle={styles.descriptionStyle}
      left={() => (
        <Image
          resizeMode="contain"
          style={{
            alignSelf: 'center',
            width: '8.5%',
            height: '100%',
            marginRight: 10,
          }}
          source={props.logo}
        />
      )}
      right={() =>
        props.showSwitch ? (
          <Switch
            value={props.isSwitchOn}
            onValueChange={props.onValueChange}
            color="#01D167"
            style={Platform.OS == 'ios' ? styles.switchStyle : null}
          />
        ) : null
      }
      onPress={props.onPress}
      rippleColor={props.showSwitch ? '#fff' : 'rgba(0, 0, 0, .05)'}
      style={{paddingLeft: 0}}
    />
  );
};

const styles = StyleSheet.create({
  titleStyle: {
    fontSize: size.font14,
    color: '#25345F',
    fontFamily: type.avenirMedium,
  },
  descriptionStyle: {
    fontSize: size.font13,
    color: '#22222266',
    fontFamily: type.avenirRegular,
    marginTop: 5,
  },
  switchStyle: {
    transform: [{scaleX: 0.8}, {scaleY: 0.8}],
    alignSelf: 'center',
  },
});

export default IconListItem;
