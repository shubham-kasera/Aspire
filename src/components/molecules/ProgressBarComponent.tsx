import React, {FC} from 'react';
import {Text, StyleSheet, View} from 'react-native';
import {ProgressBar} from 'react-native-paper';
import {size, type} from '../../theme/fonts';
import {formatAmount} from '../../utils/Helpers';

interface ProgressBar {
  title: string;
  spendings: string;
  limit: string;
  currency?: string;
}

const ProgressBarComponent: FC<ProgressBar> = props => {
  return (
    <>
      <View style={styles.rowBetween}>
        <Text style={styles.title}>{props.title}</Text>
        <View style={{flexDirection: 'row'}}>
          <Text style={[styles.title, {color: '#01D167'}]}>
            {`${props.currency}${formatAmount(props.spendings)} `}
          </Text>
          <Text style={[styles.title, {color: '#22222233'}]}>
            {`| ${props.currency}${formatAmount(props.limit)}`}
          </Text>
        </View>
      </View>
      <ProgressBar
        progress={Number(props.spendings) / Number(props.limit)}
        color={'#01D167'}
        style={styles.ProgressBar}
      />
    </>
  );
};

const styles = StyleSheet.create({
  rowBetween: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  ProgressBar: {
    height: 15,
    borderRadius: 30,
    backgroundColor: '#e5faf0',
    marginTop: '2%',
  },
  title: {
    color: '#222222',
    fontSize: size.font13,
    fontFamily: type.avenirMedium,
  },
});

export default ProgressBarComponent;
