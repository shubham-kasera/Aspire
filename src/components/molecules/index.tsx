import Header from './Header';
import BottomSheet from './BottomSheet';
import CardComponent from './CardComponent';
import IconListItem from './IconListItem';
import ProgressBarComponent from './ProgressBarComponent';

export {Header, BottomSheet, CardComponent, IconListItem, ProgressBarComponent};
