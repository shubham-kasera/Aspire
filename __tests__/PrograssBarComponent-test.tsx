/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {ProgressBarComponent} from '../src/components/molecules';

jest.mock('react-native-gesture-handler', () => {});

it('renders correctly', async () => {
  const tree = renderer
    .create(
      <ProgressBarComponent
        title="This is title"
        spendings="345"
        limit="5000"
        currency="S$"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
