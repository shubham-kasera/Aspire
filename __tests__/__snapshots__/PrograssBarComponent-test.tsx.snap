// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#222222",
          "fontFamily": "AvenirNextLTPro-Medium",
          "fontSize": 19,
        }
      }
    >
      This is title
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#222222",
              "fontFamily": "AvenirNextLTPro-Medium",
              "fontSize": 19,
            },
            Object {
              "color": "#01D167",
            },
          ]
        }
      >
        S$345 
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#222222",
              "fontFamily": "AvenirNextLTPro-Medium",
              "fontSize": 19,
            },
            Object {
              "color": "#22222233",
            },
          ]
        }
      >
        | S$5,000
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      Object {
        "busy": true,
      }
    }
    accessibilityValue={
      Object {
        "max": 100,
        "min": 0,
        "now": 6.9,
      }
    }
    accessible={true}
    onLayout={[Function]}
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#e5faf0",
          "borderRadius": 30,
          "height": 15,
          "marginTop": "2%",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    />
  </View>,
]
`;
